includes/class-wcpdf-settings-upgrade.php

```diff
@@ -98,11 +98,90 @@ public function extension_overview( $tab, $section ) {
 					'extensions' => ['templates', 'bundle'],
 				),
 			);
+			
+			$transient_name = 'wpo_wcpdf_extension_license_infos';
+			//delete_transient( $transient_name ); for debug only
+			if ( false === ( $extension_license_infos = get_transient( $transient_name ) ) ) {
+				$extension_license_infos = [];
+				foreach ( [ 'pro', 'templates' ] as $extension ) {
+					$extension_license_infos[$extension] = $this->get_extension_license_info( $extension );
+				}
+				
+				set_transient( $transient_name, $extension_license_infos, DAY_IN_SECONDS );
+			}
 
 			include( WPO_WCPDF()->plugin_path() . '/includes/views/upgrade-table.php' );
 		}
 	}
 	
+	/**
+	 * Check if a PDF extension is enabled
+	 *
+	 * @param  string  $extension  can be 'pro' or 'templates'
+	 * @return boolean
+	 */
+	public function extension_is_enabled( $extension ) {
+		$is_enabled = false;
+		
+		if ( ! empty( $extension ) || ! in_array( $extension, [ 'pro', 'templates' ] ) ) {
+			$extension_main_function = "WPO_WCPDF_".ucfirst( $extension );
+			if ( function_exists( $extension_main_function ) ) {
+				$is_enabled = true;
+			}
+		}
+		
+		return $is_enabled;
+	}
+	
+	/**
+	 * Get PDF extension license info
+	 *
+	 * @param  string  $extension  can be 'pro' or 'templates'
+	 * @return array
+	 */
+	public function get_extension_license_info( $extension ) {
+		$license_info = [
+			'enabled' => false
+		];
+		
+		if ( $this->extension_is_enabled( $extension ) ) {
+			$license_info['enabled'] = true;
```

Comment:
I don't think we should save this in the transient as it should represent the value at the time of loading the page. 

Also later in the `upgrade-table.php` view we only want to check if the extension is installed. It doesn't even has to be active. If we find either Pro or Templates is installed we can assume they have bought the extension and therefore have a license. So we can show 'Currently installed' for that extension, and display the 'Upgrade now' button for the Bundle.

Reply:
If it's only active I don't have a way to get the upgrade link because I don't know the license key.